### DISCUSSION TITLE
Meta tags are being moved to <body> when js is disabled.

### DISCUSSION THREAD

**Discussion by Coupem** [vercel/next.js#35847] [Help]
So how to save meta tags or links in tag <head> when javascript is disabled. Now on my project all meta tags or links, whitch are not in file document.js are being moved to tag <body>. On website nextjs.org  it works as expected. Thanks in advance

**Comment by icyJoseph** [comment]
Hi,

Could you please rephrase your question, and maybe provide screenshots of what behaviour you are observing when JS is disabled, and when JS is enabled?

**Reply by mike-gusev-aev** [reply]
Hi.
I have the same question about nextJs. So by default I have this meta tags on  **"head"** 
![image](https://user-images.githubusercontent.com/102963774/161535168-94deb7cb-0c0b-4c96-af23-354d35c5a085.png)

but when I disable javaScript, meta tags and links are being moved to **"body"** 

![image](https://user-images.githubusercontent.com/102963774/161535856-59955154-4be4-4179-9d69-af6258911e61.png) 

On the **nextjs.org**, all content in the **"head"** tag with javaScript and javaScript disabled remains unchanged. I need to implement the same logic

**Reply by icyJoseph** [reply]
I see, well, I am gonna guess that the `nextjs.org` is the default. I don't think any special needs to take place, I just tested this with one of my sites and the tags don't move. 

You do have a `<noscript></noscript>` tag, what's inside it? Could there be a mistake and the parser ends the head on the error?

In this application publicly available? I'd like to take a look.

**Comment by peter-peter-22** [comment]
I ecountered the same issue. It happens when there is a **suspense** component above a page and the site in in **build** mode.
When I **remove all suspenses** from my layout, the metadata moves to the head.
To do this, I have to disable the **dynamicIO** experimental feature, what demands me to use suspense. 
As a side effect, I cannot use the cache tags.

In short:
Remove all suspenses above the page.jsx files, remove dynamicIO from next.config.mjs and remove all cache tags.

This appears to be a bug and I think it's the task of nextjs to fix it.

**Reply by joacub** [reply]
yeah, is something with the page being cached that renders faster than the metadata so nextjs is pushin this to the end of the page
